applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py

```diff
@@ -0,0 +1,180 @@
+                self.output_file = TimeBasedAsciiFileWriterUtility(self.model_part,
+                                                                   file_handler_params, file_header).file
+
+    def ExecuteFinalizeSolutionStep(self):
+
+        current_time = self.model_part.ProcessInfo[KratosMultiphysics.TIME]
+        current_step = self.model_part.ProcessInfo[KratosMultiphysics.STEP]
+
+        if((current_time >= self.interval[0]) and (current_time < self.interval[1])) and (current_step % self.output_step == 0):
+            cfl_value = self._EvaluateCFL()
+
+            if (self.model_part.GetCommunicator().MyPID() == 0):
+                output = self._SummarizeCFL(cfl_value)
+                output_vals = [format(val, self.format) for val in output]
+
+                # not formatting time in order to not lead to problems with time recognition
+                # in the file writer when restarting
+                output_vals.insert(0, str(current_time))
+
+                res_labels = ["time: ", "mean: ", "std: ", "max: ", "cfl" +
+                              "{:.1f}".format(self.cfl_threshold) + ": ", "cfl1.0: "]
+
+                if (self.print_to_screen):
+
+                    result_msg = 'CFL evaluation for model part ' + \
+                        self.model_part_name + '\n'
+                    result_msg += ', '.join([a+b for a,
+                                             b in zip(res_labels, output_vals)])
+                    self._PrintToScreen(result_msg)
+
+                if (self.write_output_file):
+                    self.output_file.write(' '.join(output_vals) + "\n")
+
+    def ExecuteFinalize(self):
+        if (self.model_part.GetCommunicator().MyPID() == 0):
+            self.output_file.close()
+
+    def _GetFileHeader(self):
+        header = '# CFL for model part ' + self.model_part_name + \
+            '| CFL_threshold: ' + str(self.cfl_threshold) + '\n'
+        header += '# Time Mean Std Max HowMany>' + \
+            "{:.1f}".format(self.cfl_threshold) + ' [%] HowMany>1.0 [%]\n'
+        return header
+
+    def _PrintToScreen(self, result_msg):
+        KratosMultiphysics.Logger.PrintInfo(
+            "CFLOutputProcess", "CFL VALUE RESULTS:")
+        KratosMultiphysics.Logger.PrintInfo(
+            "CFLOutputProcess", "Current time: " + result_msg)
+
+    def _CalculateWithRespectToThreshold(self, x):
+
+        y = [val for val in x if val < self.cfl_threshold]
+        y1 = [val for val in x if val < 1.0]
+        # % of element with cfl above threshold
+        how_many = ((len(x)-len(y))/len(x))*100
+        # % of element with cfl above 1
+        how_many1 = ((len(x)-len(y1))/len(x))*100
+
+        # quantifying the mean and std for values below the threshold
+        y_mean = mean(y)
+        y_std = stdev(y)
+
+        # qunatifying the global max
+        x_max = max(x)
+        return [y_mean, y_std, x_max, how_many, how_many1]
+
+    def _EvaluateCFL(self):
```

Comment:
According to the standard, "private" methods in Python should be preceded by an `_` and written in `snake_case`.

Reply:
Actually it is correct -> @rubenzorrilla check the style guide, we agreed on 
Public: PublicMethod
Protected: _ProtectedMethod
Private: __PrivateMethod
A long time ago
It is not yet done like that everywhere, but at least for new code it is suggested to follow this

Reply:
OK then. My apologies. @mpentek forget about my comment.